### PR TITLE
Add remove

### PR DESCRIPTION
**Pull request by afizsavage**
- Add functionality to add todo
- Add functionality to edit to descriptions
- Delete single todo
- Delete all completed item

### CODE REVIEW

File: src/scripts/index.js [hunk 1]
```diff
@@ -48,49 +47,71 @@ const createAddTodoForm = () => {
...
+
+  generateTodoTemplate(todos);
   todoParent.appendChild(list);
+  addTodoForm.addEventListener('submit', (event) => {
+    event.preventDefault();
+    addNewTodo(todos);
+    generateTodoTemplate(todos);
+    storage.setItem('todos', JSON.stringify(todos));
+    updateCompletedStatus(todos);
+  });
```
Comment (rotimiazeez) [inline src/scripts/index.js]: - [ ] Good job implementing a function for adding a new task 👏, please note that you have no function for deleting a task (remove an element from the array). Kindly see to that😉

File: src/scripts/update-todos.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  event.target.classList.remove('show');
+  des.classList.remove('hide');
+  des.parentNode.parentNode.classList.remove('edit');
+  todosParam[index].description = event.target.value;
+  const yew = todosParam.filter((to) => to.description !== '');
+  storage.setItem('todos', JSON.stringify(yew));
+  generateTodo(yew);
+  event.target.parentNode.parentNode.remove();
+};
+
```
Comment (rotimiazeez) [inline src/scripts/update-todos.js]: - [ ] Please note that when I edit a todo and click enter, the new todo is not implemented and I only get the previous value back🤔

File: src/scripts/index.js (hunk 1 above)
Comment (rotimiazeez) [reply src/scripts/index.js]: - [ ] Kindly make sure a function for the "Clear all completed" button using the `Array.filter()` method is implemented.

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
# Hi @afizsavage👋, 
## Good job so far!🔥
### Summary 👏
- No linter errors ✔
- Good commit message 👍
- Descriptive PR message😊
- Good Readme 🚀
#### You still have some function to implement to go to the next stage but you are close😎🚀

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/scripts/update-todos.js [hunk 3]
```diff
@@ -11,16 +11,20 @@ class Todo {
   }
 }
 
+const deleteTodo = (allTodo, event) => {
+  const yew = allTodo.filter((to) => to.description !== '');
+  storage.setItem('todos', JSON.stringify(yew));
+  generateTodo(yew);
+  event.target.parentNode.parentNode.remove();
+};
+
```
Comment (D4nielJ) [inline src/scripts/update-todos.js]: - [ ] I think the delete event is not clear enough for the users. Let's create a button with a trash icon. When this button is pressed the task should be removed both from the screen and the array of items in the localStorage. With the current behavior whenever I select a task and then click out of it gets deleted.

- [ ] Let's try to clean the list from bugs as well. After I delete a task and I tried to add a new one, empty artifacts are created in the list. Watch the following clip for reference:

todo_list_with_bugs

[OPTIONAL] As you can see in the clip, I have to delete each task from the input before adding another one. I think you could improve the User Experience if you clean the input after each submit event.

**Review by D4nielJ — CHANGES_REQUESTED** [review]
## Hello @afizsavage, 

Great work on making the changes requested by the previous reviewer(s) 👏

You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## Happy coding and stay awesome! 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can always find me on slack!

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/scripts/update-todos.js (hunk 3 above)
Comment (afizsavage) [reply src/scripts/update-todos.js]: Hello @D4nielJ :smile: 

I have implemented the required changes. I've added the trash icon to the input field to show the user that she can delete a to-do item while editing.

I made a short screencast of how the original todo app that we are supposed to clone implement the feature.
If you can please see the screencast at the following [screencast](https://drive.google.com/file/d/1RPFf4xdfiQW6BS_TX_46kss9PFyERcrm/view?usp=sharing)

Comment (henatan99) [reply src/scripts/update-todos.js]: Things are well explained clarified. Well done! :clap: :clap: :clap:

**Review by henatan99 — APPROVED** [review]
Hi @afizsavage , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**